Домашнее задание 4 чат на JavaFX

Kirikx opened the pull request.

Kirikx committed:
Домашнее задание 4 чат на JavaFX

OKrylov commented on src/Chat/Controller.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    private void messageStack(Event e) {
+        Node source = (Node) e.getSource();
```
Корректный подход, но проще объявить поля TextField и TextArea, указав в fxml их fx:id, что должны совпадать с именами полей. Тогда мы можем обращаться к ним напрямую, не ища их через scene

OKrylov reviewed (approved):
Все отлично, задача выполнена корректно

Kirikx committed:
Домашнее задание 4 чат на JavaFX в MAVEN

Kirikx committed:
Движемся вперед

Kirikx committed:
HW7
1. Разобраться с кодом. Done!
2. *Реализовать личные сообщения так: если клиент пишет «/w nick3 Привет», то только клиенту с ником nick3 должно прийти сообщение «Привет». Done!
NW8
1. Разобраться с кодом. Done!
2. Добавить отключение неавторизованных пользователей по таймауту (120 сек. ждем после подключения клиента. Если он не авторизовался за это время, закрываем соединение) Done!

Kirikx committed:
1. Внедрена БД SQLite/
2. Добавлена аутентификация через БД

Kirikx committed:
1. Добавлена возможность смены ника при отправке сообщения в формате "/rename login password"

Kirikx committed:
1. Добавить в сетевой чат запись локальной истории в текстовый файл на клиенте. Done!
2. После загрузки клиента показывать ему последние 100 строк чата. Done!

3. Добавлена автозамена имени в переписке

Kirikx committed:
1. Добавлено логирование класса MyServer loj4j
